common/include/pcl/exceptions.h

```diff
@@ -262,4 +262,17 @@ namespace pcl
                           unsigned line_number = 0)
       : pcl::PCLException (error_description, file_name, function_name, line_number) { }
   };
+
+  /** \class OutofMemoryException
+    * \brief An exception that is thrown when out of memory.
+    */
+  class OutOfMemoryException : public PCLException
```

Comment:
[c++ has one](https://en.cppreference.com/w/cpp/memory/new/bad_alloc), which I'd try to catch. It'd be better if you inherit from both (somehow, it might cause a diamond inheritance, just something to watch out for)

Reply:
Kunal brings a good point into the table. PCLException derives from std::runtime_error and we're clearly handling a bad_aloc situation. The quickest fix for me at the moment would be to throw a simple bad_aloc, cheap and dirty. We can address the limitations of PCLException in another issue and this one would be fixed.

Reply:
Should be changed now. I have added a PCL_ERROR just before, to give a bit of help figure out where the exception throws.